client/src/controllers/TeleportController.ts

```diff
@@ -40,6 +45,7 @@ export class TeleportController extends Controller<HTMLTemplateElement> {
 
     const complete = () => {
       if (completed) return;
+      if (this.resetValue) target.innerHTML = '';
```

Comment:
This is great.

Maybe the `append` event name should change. 

Additionally, it would be great to see a unit test.

For the sake of time though, these are not blockers.

I'm happy to prepare an issue for someone to contribute these later.

Reply:
@lb- I believe we’re intending to do further refactorings & add tests once the dust settles on Wagtail 6.0